setup.py

```diff
@@ -1,20 +1,28 @@
 # pylint: disable=not-callable, no-member, invalid-name, line-too-long, wildcard-import, unused-wildcard-import, missing-docstring
 from setuptools import setup, find_packages
-from torch.utils.cpp_extension import BuildExtension, CUDAExtension
+import torch
+from torch.utils.cpp_extension import BuildExtension, CUDAExtension, CUDA_HOME
 
 # python setup.py develop    - if you wont to be able to execute from PyCharm (or similar IDE) - places .so file into se3cnn folder from which real_spherical_harmonics imports
 
 # Or:
 # python setup.py build_ext
 # python setup.py install    - PyCharm won't work, because it can't resolve import, but executable from terminal
 
-ext_modules = [
-    CUDAExtension('se3cnn.real_spherical_harmonics',
-                  sources=['src/real_spherical_harmonics/rsh_bind.cpp',
-                           'src/real_spherical_harmonics/rsh_cuda.cu'],
-                  extra_compile_args={'cxx': ['-std=c++14'],
-                                      'nvcc': ['-std=c++14']})
-]
+if not torch.cuda.is_available():
+    ext_modules = None
+    print("GPU is not available. Skip building CUDA extensions.")
+elif torch.cuda.is_available() and CUDA_HOME is not None:
+    ext_modules = [
+        CUDAExtension('se3cnn.real_spherical_harmonics',
+                      sources=['src/real_spherical_harmonics/rsh_bind.cpp',
+                               'src/real_spherical_harmonics/rsh_cuda.cu'],
+                      extra_compile_args={'cxx': ['-std=c++14'],
+                                          'nvcc': ['-std=c++14']})
+    ]
+else:
+    # GPU is available, but CUDA_HOME is None
+    raise AssertionError("CUDA_HOME is undefined. Make sure nvcc compiler is available (cuda toolkit installed?)")
```

Comment:
Maybe it's worth raising the error/warning from pytorch itself in addition to this clarifying information?

Reply:
What exactly do you mean? (It does raise error)

Reply:
A user might want to google the error. This error does not yield results if you google the error since it is custom written. However, there is a pytorch error which is raised when pytorch cannot find nvcc. I suggest raising this error either instead or too.

Reply:
So, the error in question is: 
"OSError: CUDA_HOME environment variable is not set. Please set it to your CUDA install root."
In principle, we can append the text, and change type of the raised error. 
Though personally I do not think this is particularly useful. Joke you not (really, try to search for it), Google provides quite a few ways to resolve more or less edgy issues with paths, but does not convey the fact that you need specific compiler in the first place, and that you get it with cuda-toolkit (which sets paths automatically).

Reply:
True. When I google that one, I don't find answers. Okay, your error is better!